integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/cache/ESBJAVA4721PIWithCacheTestCase.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2017 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License, 
+ * Version 2.0 (the "License"); you may not use this file except 
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.esb.mediator.test.cache;
+
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.automation.engine.exceptions.AutomationFrameworkException;
+import org.wso2.carbon.automation.test.utils.http.client.HttpRequestUtil;
+import org.wso2.carbon.automation.test.utils.http.client.HttpResponse;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+import org.wso2.esb.integration.common.utils.common.ServerConfigurationManager;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This test the ability of JSON to XML conversion with processing instructions enabled when caching is enabled.
+ */
+public class ESBJAVA4721PIWithCacheTestCase extends ESBIntegrationTest {
+
+    private ServerConfigurationManager serverConfigurationManager;
+
+    @BeforeClass(alwaysRun = true)
+    public void deployArtifacts() throws Exception {
+        super.init();
+        serverConfigurationManager = new ServerConfigurationManager(context);
+        serverConfigurationManager.applyConfiguration(new File(
+                getESBResourceLocation() + File.separator + "mediatorconfig" + File.separator + "cache" + File.separator
+                        + "synapse.properties"));
+        super.init();
+        loadESBConfigurationFromClasspath(
+                File.separator + "artifacts" + File.separator + "ESB" + File.separator + "mediatorconfig"
+                        + File.separator + "cache" + File.separator + "CacheMediatorforJasonResponse.xml");
+    }
+
+    @Test(groups = "wso2.esb",
+          description = "Test cache mediator with  Json response having a single element array with PI enabled")
+    public void testJsonResponseWithCacheMediator() throws IOException, AutomationFrameworkException {
```

Comment:
Even if we throw the sub classes in test methods, no one will handle those exceptions explicitly. It is handled by TestNG and TestNG marked them error when an exception is thrown from a test method and we can see the actual exception class in test reports. 
Throwing super class also make sure that the test method will not be broken even exception classes is changed in the method which is used in test methods.

It's not mandatory, but the usual practice we follow when writing test cases.